tools/tpm2_createprimary.c

```diff
@@ -91,17 +96,20 @@ static bool set_name_alg(TPMI_ALG_HASH halg, TPM2B_PUBLIC *public) {
 
 static bool set_alg(TPMI_ALG_PUBLIC type, TPM2B_PUBLIC *public) {
 
-
     switch(type) {
     case TPM2_ALG_RSA: {
         TPMS_RSA_PARMS *r = &public->publicArea.parameters.rsaDetail;
-       r->symmetric.algorithm = TPM2_ALG_AES;
-       r->symmetric.keyBits.aes = 128;
-       r->symmetric.mode.aes = TPM2_ALG_CFB;
-       r->scheme.scheme = TPM2_ALG_NULL;
-       r->keyBits = 2048;
-       r->exponent = 0;
-       public->publicArea.unique.rsa.size = 0;
+        if (public->publicArea.objectAttributes & TPMA_OBJECT_SIGN_ENCRYPT) {
+            r->symmetric.algorithm = TPM2_ALG_NULL;
```

Comment:
Do you also need `r->symmetric.scheme = TPM_ALG_NULL` here? `12.2.3.5 TPMS_RSA_PARMS` in `Trusted Platform Module Library Part 2: Structures` says:

```
NOTE   When both sign and decrypt are SET, restricted shall be CLEAR
       and scheme shall be TPM_ALG_NULL.
```

Reply:
I'm not sure. The simulator didn't kick back what I sent